src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -201,16 +228,45 @@ export class UiSettingsClient implements IUiSettingsClient {
     return filteredValues;
   }
 
+  /**
+   * group change into different scopes
+   * @param changes ui setting changes
+   * @returns [global, user]
+   */
+  private groupChanges(changes: Record<string, any>) {
+    const userLevelKeys = [] as string[];
+    Object.entries(this.defaults).forEach(([key, value]) => {
+      if (value.scope === 'user' || (Array.isArray(value.scope) && value.scope.includes('user'))) {
+        userLevelKeys.push(key);
+      }
+    });
+    const userChanges = {} as Record<string, any>;
+    const globalChanges = {} as Record<string, any>;
+
+    Object.entries(changes).forEach(([key, val]) => {
+      if (userLevelKeys.includes(key)) {
+        userChanges[key] = val;
+      } else {
+        globalChanges[key] = val;
+      }
+    });
+
+    return [globalChanges, userChanges];
+  }
+
   private async write({
     changes,
     autoCreateOrUpgradeIfMissing = true,
+    userLevel = false,
```

Comment:
I think a boolean flag `userLevel: boolean` may not flexible enough, imagine in the future, we may have uiSettings client to support workspace level settings.

Reply:
update to scope as suggested, it's more flexible. thanks.